src/main/java/teammates/logic/api/EmailGenerator.java

```diff
@@ -250,6 +258,76 @@ private EmailWrapper generateSubmissionConfirmationEmail(
 
     }
 
+    private EmailWrapper generateLinkRecoveryEmailForUser(String userEmail) {
+        Instant endTime = Instant.now();
+        Instant startTime = endTime.minus(Duration.ofDays(180));
+        String subject = EmailType.FEEDBACK_ACCESS_LINKS_RESENT.getSubject();
+        StringBuffer linksFragmentValue = new StringBuffer(1000);
+        String teammateHomePageLink = Config.APP_URL;
+        String emailBody;
+        String studentName = null;
+
+        List<FeedbackSessionAttributes> sessions = fsLogic.getAllFeedbackSessionsWithinTimeRange(startTime, endTime);
+        for (FeedbackSessionAttributes session : sessions) {
+            CourseAttributes course = coursesLogic.getCourse(session.getCourseId());
```

Comment:
course might be null...

Reply:
In my logic an empty list will be returned in that case.

Reply:
>In my logic an empty list will be returned in that case.

i m a bit confused too...

Reply:
Yes I consulted @xpdavid about this, The only case course will be null  is that cascading has not finish yet, Which is an event with extremely low probability. So We don't consider that case in here.